controllers/front/product.php

```diff
@@ -45,16 +45,13 @@ public function initContent()
         if ($token === false) {
             exit();
         }
-
+        header('Content-type: text/xml');
+        header('Etag: ' . ShoppingfeedPreloading::getEtag($token['id_shoppingfeed_token']));
```

Comment:
Could you please read the etag specification : https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag ?
Where is the If-Match/If-None-Match request management ? You should response 304 without computing and returnig the body if anything change.